### DISCUSSION TITLE
Broadcast size

### DISCUSSION THREAD

**Discussion by DiegoMartinezRdg** [cesanta/mongoose#2026] [Ask questions, get answers] [answered]
When I send a json string of 165 characters , using broadcast , the broadcast server receives a diferent (and random) size on the io buffer len. 
Is always greater than what I sent, but it fills the data with a tail of garbage. 

What I am doing wrong?

```
//---- receive
static void ev_handler(struct mg_connection *nc, int ev, void *p) 
{
	struct mbuf *io = &nc->recv_mbuf;		
	switch (ev) 
	{	
	case MG_EV_SEND:
		nc->flags |= MG_F_SEND_AND_CLOSE;
		mbuf_remove(io, io->len);		
		break;
	case MG_EV_RECV:
		{
		mbuf_remove(io, io->len);
		std::string ip=inet_ntoa(nc->sa.sin.sin_addr);		
                std::string receive((unsigned char*)io->buf, (int)io->size);
		nc->recv_mbuf.len = 0;
		}
		break;		
	default: break;
	}
}
```

**Comment by cpq** [comment, marked as answer]
@DiegoMartinezRdg here is a list of wrong things:
1. You're using an old version we provide no support for. Please update.
2. You did a bad job explaining what you're doing. Instead, file an issue and follow the template to provide all necessary context. Of course, after you've updated to the recent version.
